system/cpu.go

```diff
@@ -0,0 +1,38 @@
+package system
+
+import (
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+type cpu struct{}
+
+// CPU is a singleton that returns information about the CPU
+var CPU cpu
+
+// Architecture returns the CPU's architecture name as a string
+func (cpu) Architecture() (string, error) {
+	var utsname unix.Utsname
+	err := unix.Uname(&utsname)
+	if err != nil {
+		return "", err
+	}
+
+	architecture := string(utsname.Machine[:])
+	architecture = strings.Trim(architecture, "\x00")
```

Comment:
The value returned from `utsname.Machine[:]` is a string with a bunch of `NUL` values after it for some reason. So this strips those off the end.